var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -178,9 +178,15 @@ class Llvm(CMakePackage, CudaPackage):
     # MLIR exists in > 10.x
     conflicts("+mlir", when="@:9")
 
+
+    # code signing is only necessary on macOS",
+    conflicts('+code_signing', when='platform=linux')
+    conflicts('+code_signing', when='platform=bgq')
+    conflicts('+code_signing', when='platform=cray')
```

Comment:
Is there any way to do:
```
conflicts('+code_signing', when='platform!=darwin')
```
???

If not, should we maybe add that functionality to core Spack for this PR?

Reply:
As per https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/apple-libunwind/package.py#L16, no. Is that comment still valid ?

Reply:
That comment is still valid. There is no "not" in Spack's spec syntax, and it would be non-trivial to add one.

Reply:
Is there a list or tuple of platforms somewhere so it could be:

```python
for p in (plat if plat != 'darwin' for plat in spack.somehow.get.platforms):
  conflicts('+code_signing', when=f'platform={plat}')
```

Or similar?

Reply:
I wish, see #14717

Reply:
Dang... well, was worth asking but I guess this is what we're stuck with for now.